### PR TITLE
Remove legacy .apk build.

### LINKED ISSUE #8455: Discontinue non-gradle build
`flutter create` and all examples have switched to building with gradle. 

This tracks finally the support for non-gradle builds.

### LINKED ISSUE #8684: Deleting the app on your device leaves you unable to use `flutter run`
If you run `flutter run`, then delete the app from your device using the home screen app list, then run `flutter run` again, the app doesn't run, because the sha hasn't been removed from the device and thus we think we still have the same app on your device.

### PR DESCRIPTION
**Pull request by jakobr-google**
Print out an error message telling the user to upgrade the project if
it's not Gradle-based. Removed all the obvious traces of the legacy
build.

Added support for Dart VM kernel snapshots in Gradle builds.

Fixed Android installs to verify that the app is actually installed, and
not just rely on the presence of the .sha1 file.

Fixes #8455 
Fixes #8684

### CODE REVIEW

**Comment by jakobr-google** [comment]
This is the text that's printed out when you try to `flutter build apk` or `flutter run` an old project:

```
Launching lib/main.dart on Android SDK built for x86 64 in debug mode...
The build process for Android has changed, and the current project configuration
is no longer valid. Please consult

  https://github.com/flutter/flutter/wiki/Upgrading-Flutter-projects-to-build-with-gradle

for details on how to upgrade the project.
```

**Comment by abarth** [comment]
LGTM but I'm not 100% sure about the kernel stuff.

**Comment by jakobr-google** [comment]
I'm not 100% sure about the kernel stuff either, but I've now fixed it so it seems to work. Except I don't know how to build a kernel snapshot. I'm hoping @danrubel will help test it once it's landed.

**Comment by danrubel** [comment]
The kernel based changes look good. I'm happy to fix any issues that crop up in that area in subsequent PRs.

File: packages/flutter_tools/lib/src/android/gradle.dart [hunk 1]
```diff
@@ -197,6 +197,9 @@ Future<Null> buildGradleProjectV2(String gradle, String buildModeName, String ta
   if (target != null) {
     command.add('-Ptarget=$target');
   }
+  if (kernelPath != null) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/android/gradle.dart]: nit: no braces around single line statements

File: packages/flutter_tools/lib/src/flx.dart [hunk 2]
```diff
@@ -124,9 +93,14 @@ Future<Null> build({
     snapshotFile = fs.file(snapshotPath);
   }
 
+  DevFSContent kernelContent;
+  if (kernelPath != null) {
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/flx.dart]: Ditto

**Pull request merged**